cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -128,6 +145,161 @@ func createKubeDNSAddon(deploymentBytes, serviceBytes []byte, client clientset.I
 	return nil
 }
 
+func coreDNSAddon(cfg *kubeadmapi.MasterConfiguration, client clientset.Interface) error {
+	k8sVersion, err := version.ParseSemantic(cfg.KubernetesVersion)
```

Comment:
move the k8sVersion parsing here to `EnsureDNSAddon` and pass to both these functions (coreDNSAddon and kubeDNSAddon)